### DISCUSSION TITLE
Why bundler starts to download all specs for the gem

### DISCUSSION THREAD

**Discussion by hurricup** [ruby/rubygems#7717] [Q&A]
I have a system ruby on ubuntu 22.04 with default bundler 2.22

```
Bundler version 2.2.22

RubyGems Environment:
  - RUBYGEMS VERSION: 3.3.5
  - RUBY VERSION: 3.0.2 (2021-07-07 patchlevel 107) [x86_64-linux-gnu]
  - INSTALLATION DIRECTORY: /var/lib/gems/3.0.0
  - USER INSTALLATION DIRECTORY: /home/user/.local/share/gem/ruby/3.0.0
  - RUBY EXECUTABLE: /usr/bin/ruby3.0
  - GIT EXECUTABLE: /usr/bin/git
  - EXECUTABLE DIRECTORY: /usr/local/bin
  - SPEC CACHE DIRECTORY: /home/user/.local/share/gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /etc
  - RUBYGEMS PLATFORMS:
     - ruby
     - x86_64-linux
  - GEM PATHS:
     - /var/lib/gems/3.0.0
     - /home/user/.local/share/gem/ruby/3.0.0
     - /usr/local/lib/ruby/gems/3.0.0
     - /usr/lib/ruby/gems/3.0.0
     - /usr/lib/x86_64-linux-gnu/ruby/gems/3.0.0
     - /usr/share/rubygems-integration/3.0.0
     - /usr/share/rubygems-integration/all
     - /usr/lib/x86_64-linux-gnu/rubygems-integration/3.0.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /home/user/.cargo/bin
     - /usr/local/sbin
     - /usr/local/bin
     - /usr/sbin
     - /usr/bin
     - /sbin
     - /bin

```

I have a  simple `Gemfile` :
```
source 'http://rubygems.org'

gem 'cucumber'
```

And i need to `bundle install` it. This is a scenario from one of tests. And it worked fine until today morning. 

But now  it looks like for some reason bundler starts to download all cucumber specs, and dies on https://github.com/cucumber/cucumber-ruby/issues/136

Question is why and is it possible to avoid such issues.

[bundlefail.log](https://github.com/user-attachments/files/15604679/bundlefail.log)

I was unable to reproduce this on my local machine or in the same docker container test scenario runs. Probably there are some infrastructural issues on the test host machine.

**Comment by deivid-rodriguez** [comment]
For some reason the compact index API requests timed out. In that case, Bundler falls back to fetching the full index. It was most likely a temporary problem.

**Reply by hurricup** [reply]
I see. But note, that this may cause a problem when some old version of a gem was broken (broken gemspec), bundler unable to skip it and fails. When it may be shouldn't.
